ignite/cmd/bubblemodel/testnet_multi_node.go

```diff
@@ -0,0 +1,233 @@
+var _ tea.Model = MultiNode{}
+
+// MultiNode represents a set of nodes, managing state and information related to them.
+type MultiNode struct {
+	appd string
+	args chain.MultiNodeArgs
+
+	nodeStatuses []NodeStatus
+	pids         []int      // Store the PIDs of the running processes
+	numNodes     int        // Number of nodes
+	logs         [][]string // Store logs for each node
+}
+
+// ToggleNodeMsg is a structure used to pass messages
+// to enable or disable a node based on the node index.
+type ToggleNodeMsg struct {
+	nodeIdx int
+}
+
+// UpdateStatusMsg defines a message that updates the status of a node by index.
+type UpdateStatusMsg struct {
+	nodeIdx int
+	status  NodeStatus
+}
+
+// UpdateLogsMsg is for continuously updating the chain logs in the View.
+type UpdateLogsMsg struct{}
+
+// UpdateDeemon returns a command that sends an UpdateLogsMsg.
+// This command is intended to continuously refresh the logs displayed in the user interface.
+func UpdateDeemon() tea.Cmd {
+	return func() tea.Msg {
+		return UpdateLogsMsg{}
+	}
+}
+
+// NewModel initializes the model.
+func NewModel(chainname string, args chain.MultiNodeArgs) (MultiNode, error) {
+	numNodes, err := strconv.Atoi(args.NumValidator)
+	if err != nil {
+		return MultiNode{}, err
+	}
+	return MultiNode{
+		appd:         chainname + "d",
+		args:         args,
+		nodeStatuses: make([]NodeStatus, numNodes), // initial states of nodes
+		pids:         make([]int, numNodes),
+		numNodes:     numNodes,
+		logs:         make([][]string, numNodes), // Initialize logs for each node
+	}, nil
+}
+
+// Init implements the Init method of the tea.Model interface.
+func (m MultiNode) Init() tea.Cmd {
+	return nil
+}
+
+// ToggleNode toggles the state of a node.
+func ToggleNode(nodeIdx int) tea.Cmd {
+	return func() tea.Msg {
+		return ToggleNodeMsg{nodeIdx: nodeIdx}
+	}
+}
+
+// RunNode run or stop the node based on its status.
+func RunNode(nodeIdx int, start bool, m MultiNode) tea.Cmd {
+	var (
+		pid  = &m.pids[nodeIdx]
+		args = m.args
+		appd = m.appd
+	)
+
+	return func() tea.Msg {
+		if start {
+			nodeHome := filepath.Join(args.OutputDir, args.NodeDirPrefix+strconv.Itoa(nodeIdx))
+			// Create the command to run in background as a daemon
+			cmd := exec.Command(appd, "start", "--home", nodeHome)
+
+			// Start the process as a daemon
+			cmd.SysProcAttr = &syscall.SysProcAttr{
+				Setpgid: true, // Ensure it runs in a new process group
+			}
+
+			stdout, err := cmd.StdoutPipe() // Get stdout for logging
+			if err != nil {
+				fmt.Printf("Failed to start node %d: %v\n", nodeIdx+1, err)
+				return UpdateStatusMsg{nodeIdx: nodeIdx, status: Stopped}
+			}
+
+			err = cmd.Start() // Start the node in the background
+			if err != nil {
+				fmt.Printf("Failed to start node %d: %v\n", nodeIdx+1, err)
+				return UpdateStatusMsg{nodeIdx: nodeIdx, status: Stopped}
+			}
+
+			*pid = cmd.Process.Pid // Store the PID
+			go func() {
```

Comment:
Let's use an errogroup or cancel context to close the routine if the context is dead or canceled. We can pass the context (`cmd.Context()`)  to the `RunNode` method

```go
func RunNode(ctx context.Context, nodeIdx int, start bool, m MultiNode) tea.Cmd {
```

Reply:
If the start is not correct or anything causes the chain to stop, it will also be shown on the chain log.

For example, the default address is already in use at another node.
<img width="905" alt="image" src="https://github.com/user-attachments/assets/9cbb285b-67e6-4d92-962e-965493d5bd94">

Reply:
But do we need to deploy all nodes together if we change the port?

Reply:
Other nodes are still running normally.

<img width="1010" alt="image" src="https://github.com/user-attachments/assets/5b91ce57-31d8-47a3-9596-c95ad07602b2">

Reply:
And can we pass the context to this function and use it to kill this uncontrolled goroutine?

Reply:
To exit a node x, just press the x key. x= 1..9

> I got it, but do we need to deploy all nodes together if we change the port? If yes, we must throw exit 1 to the command and abort.

Reply:
Deploying five nodes with one not working doesn't make sense. We need to break all command